### PR TITLE
Be able to get jobs using cibyl from elasticsearch

### PR DESCRIPTION
**Pull request by adrianfusco**
- Modify source registry to add the correct elasticsearch source for queries.
- Add get_jobs method. Now it returns every job it matchs in a 'AttributeDictValue'
- TODO: `get_builds` and `get_tests` methods. Tests.

### CODE REVIEW

File: cibyl/sources/elasticsearch/api.py [hunk 1]
```diff
@@ -81,9 +81,52 @@ def __query_get_hits(self: object, query: dict, index: str = '') -> list:
...
+            elasticsearch server
+            :rtype: :class:`AttributeDictValue`
+        """
+        jobs_found = self.get_jobs(**kwargs)
+
+        for job_name, job in jobs_found.items():
+            query_body = QueryTemplate('job_name', job_name).get
+
+            builds = self.__query_get_hits(
+                index='jenkins_builds',
```
Comment (bregman-arie) [inline cibyl/sources/elasticsearch/api.py:106]: can this be index=self.index? so users can modify it from configuration

Comment (adrianfusco) [reply cibyl/sources/elasticsearch/api.py:106]: It depends how we handle it. 

The thing is we’re using in one method one index (`get_jobs` with Jenkins) and in the other method another different index (`get_builds` with jenkins_builds).

Comment (bregman-arie) [reply cibyl/sources/elasticsearch/api.py:106]: Yeah so probably no. But I wish we didn't have to include indexes names hardcoded in the code

Comment (adrianfusco) [reply cibyl/sources/elasticsearch/api.py:106]: Maybe we can pass in the configuration file `jobs_index` and `builds_index` and we’ll be able to receive it as you commented. 

I’ll also think another options.

Comment (bregman-arie) [reply cibyl/sources/elasticsearch/api.py:106]: maybe. but will we need also "tests_index"?
why do we actually have two indexes for builds and jobs? can't we just use one?

Comment (adrianfusco) [reply cibyl/sources/elasticsearch/api.py:106]: Jobs and builds are in different indexes. The relationship with them is the job name. 
From one side I can get all the jobs from one index. For being able to get the builds I search them by the job name. That's why I need to specify the index for every search.

Comment (bregman-arie) [reply cibyl/sources/elasticsearch/api.py:106]: Looking at "builds_index" I see for every document the fields: "job_name", "build_result", ... why can't use only "builds_index"?

**Comment by adrianfusco** [comment]
- Added elasticsearch source in source_factory
- Added multiple query templates:
   * Now we can get one job, multiple jobs, all jobs

Still WIP.

File: cibyl/sources/elasticsearch/api.py (hunk 1 above)
Comment (bregman-arie) [reply cibyl/sources/elasticsearch/api.py:109]: k, we can do index optimization later on. Let's merge it when CI passes

Comment (adrianfusco) [reply cibyl/sources/elasticsearch/api.py:109]: Ok, I'll finish it.

**Comment by adrianfusco** [comment]
- Added method `get_last_build`.

**Comment by adrianfusco** [comment]
You can review it. There’s another task to add the tests.

**Comment by adrianfusco** [comment]
Added:

Tests for ElasticSearchOSP class. Methods: get_builds, get_jobs
Tests for QueryTemplate class.

File: cibyl/sources/elasticsearch/api.py [hunk 2]
```diff
@@ -15,58 +15,59 @@
...
+from cibyl.models.ci.build import Build
+from cibyl.models.ci.job import Job
+from cibyl.sources.elasticsearch.client import ElasticSearchClient
+from cibyl.sources.source import Source
 
 LOG = logging.getLogger(__name__)
 
 
-class ElasticSearchOSP:
+class ElasticSearchOSP(Source):  # pylint: disable=too-few-public-methods
```
Comment (bregman-arie) [inline cibyl/sources/elasticsearch/api.py:30]: Jorge has set "min-public-methods=1" in .pylintrc so you probably don't need to disable it

**Review by bregman-arie — APPROVED** [review]
Amazing job on this one :)

File: cibyl/sources/elasticsearch/api.py (hunk 2 above)
Comment (adrianfusco) [reply cibyl/sources/elasticsearch/api.py:30]: You're right. Thx.

**Pull request merged**